src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts

```diff
@@ -0,0 +1,109 @@
+import express from "express"
+import request from "supertest"
+
+import { attachReadRouteHandlerWrapper } from "@middleware/routeHandler"
+
+import { NotificationsRouter as _NotificationsRouter } from "@routes/v2/authenticatedSites/notifications"
+
+import { generateRouter } from "@fixtures/app"
+import { mockSiteName, mockIsomerUserId } from "@fixtures/sessionData"
+import NotificationsService from "@services/identity/NotificationsService"
+
+describe("Notifications Router", () => {
+  const mockNotificationsService = {
+    listRecent: jest.fn(),
+    listAll: jest.fn(),
+    markNotificationsAsRead: jest.fn(),
+  }
+
+  const NotificationsRouter = new _NotificationsRouter({
+    notificationsService: (mockNotificationsService as unknown) as NotificationsService,
+  })
+
+  const subrouter = express()
+
+  // We can use read route handler here because we don't need to lock the repo
+  subrouter.get(
+    "/:siteName/notifications/",
+    attachReadRouteHandlerWrapper(NotificationsRouter.getRecentNotifications)
+  )
+  subrouter.get(
+    "/:siteName/notifications/allNotifications",
+    attachReadRouteHandlerWrapper(NotificationsRouter.getAllNotifications)
+  )
+  subrouter.post(
+    "/:siteName/notifications/",
+    attachReadRouteHandlerWrapper(NotificationsRouter.markNotificationsAsRead)
+  )
+
+  const app = generateRouter(subrouter)
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("getRecentNotifications", () => {
+    it("should retrieve the recent notifications for a user and site", async () => {
+      // Arrange
+      const mockNotificationsValue: never[] = []
+      mockNotificationsService.listRecent.mockResolvedValueOnce(
+        mockNotificationsValue
+      )
+
+      // Act
+      const resp = await request(app)
+        .get(`/${mockSiteName}/notifications/`)
+        .expect(200)
+
+      // Assert
+      expect(resp.body).toStrictEqual(mockNotificationsValue)
+      expect(mockNotificationsService.listRecent).toHaveBeenCalledWith({
+        siteName: mockSiteName,
+        userId: mockIsomerUserId,
+      })
+    })
+  })
+
+  describe("getAllNotifications", () => {
+    it("should retrieve the recent notifications for a user and site", async () => {
```

Comment:
see above comment wrt changing description to reflect that we are calling the underlyign

Reply:
https://github.com/isomerpages/isomercms-backend/pull/514/commits/bb3cb2e4eb0dd389731cc596133dabb4eef8b063